### PR TITLE
Remove protectGames dead code

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205504063854057/f

### Description
Remove `protectGames` feature flag and associated (dead) code

### Steps to test this PR
Smoke test AppTP
Ensure that games are protected by default when AppTP is enabled.

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3589** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3589" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3589?utm_source=stack-comment).

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/apps/NewAppBroadcastReceiver.kt [hunk 1]
```diff
@@ -95,10 +93,6 @@ class NewAppBroadcastReceiver @Inject constructor(
         kotlin.runCatching { applicationContext.unregisterReceiver(this) }
     }
 
-    private fun isGame(packageName: String): Boolean {
-        return appCategoryDetector.getAppCategory(packageName) is AppCategory.Game && !appTpFeatureConfig.isEnabled(AppTpSetting.ProtectGames)
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/apps/NewAppBroadcastReceiver.kt:99]: `appCategoryDetector` and `appTpFeatureConfig` can be removed from the constructor.

Comment (aitorvs) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/apps/NewAppBroadcastReceiver.kt:99]: ah nice, will do

**Review by anikiki — APPROVED** [review]
LGTM! :tada: Smoke tested AppTP, looks good.

**Pull request merged**